### ISSUE TITLE
LT05 | Unexpected exception: '<=' not supported between instances of 'str' and 'int'

### ISSUE THREAD

**Issue by food-spotter** [sqlfluff/sqlfluff#5465]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When linting a **very simple dbt model** (dialect: snowflake, templater: jinja) an unexpected error is raised. The rule that triggers this is LT05

### Expected Behaviour

No error should be raised

### Observed Behaviour

Error raised:

```sh
$ sqlfluff lint models/staging/reference_data/stg_reference_data__country_codes.sql 

CRITICAL   [LT05] Applying rule LT05 to 'models/staging/reference_data/stg_reference_data__country_codes.sql' threw an Exception: '<=' not supported between instances of 'str' and 'int' 

Traceback (most recent call last):
  File ".venv/lib/python3.10/site-packages/sqlfluff/core/rules/base.py", line 507, in crawl
    res = self._eval(context=context)
  File ".venv/lib/python3.10/site-packages/sqlfluff/rules/layout/LT05.py", line 37, in _eval
    .break_long_lines()
  File ".venv/lib/python3.10/site-packages/sqlfluff/utils/reflow/sequence.py", line 610, in break_long_lines
    elements, length_results = lint_line_length(
  File ".venv/lib/python3.10/site-packages/sqlfluff/utils/reflow/reindent.py", line 2020, in lint_line_length
    if line_length_limit <= 0:
TypeError: '<=' not supported between instances of 'str' and 'int'
== [models/staging/reference_data/stg_reference_data__country_codes.sql] FAIL                                                                                            
L:   1 | P:   1 | LT05 | Unexpected exception: '<=' not supported between
                       | instances of 'str' and 'int';
Could you open an issue at
                       | https://github.com/sqlfluff/sqlfluff/issues ?
You can
                       | ignore this exception for now, by adding '-- noqa: LT05'
                       | at the end
of line 1
 [layout.long_lines]
All Finished 📜 🎉!
```

### How to reproduce

Sample dbt sql file:

```sql
with country_codes as (

    select

        country_name,
        country_iso2,
        country_iso3,
        country_numeric_code

    from

        {{ source('reference_data', 'raw_reference_data__country_codes') }}

)

select * from country_codes
```

Then run command:

```sh
sqlfluff lint file.sql
```

### Dialect

snowflake

### Version

sqlfluff, version 2.3.5

Python 3.10.7

### Configuration

```md
# adopting dbt cloud's rules: https://docs.getdbt.com/docs/cloud/dbt-cloud-ide/lint-format#customize-linting

[sqlfluff]
# Changing templater to dbt will make linting slower because linting will first compile dbt code into data warehouse code.
templater = jinja  
dialect = snowflake
runaway_limit = 10
max_line_length = 200  # override dbt default of 80
indent_unit = space

[sqlfluff:indentation]
tab_space_size = 4

[sqlfluff:layout:type:comma]
spacing_before = touch
line_position = trailing

[sqlfluff:rules:capitalisation.keywords] 
capitalisation_policy = lower

[sqlfluff:rules:aliasing.table]
aliasing = explicit

[sqlfluff:rules:aliasing.column]
aliasing = explicit

[sqlfluff:rules:aliasing.expression]
allow_scalar = False

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower

[sqlfluff:rules:ambiguous.column_references]  # Number in group by
group_by_and_order_by_style = implicit

```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by yoichi** [comment]
The error is caused by
```
line_length_limit='200  # override dbt default of 80'
```

@food-spotter  In your configuration file, replace
```
max_line_length = 200  # override dbt default of 80
```

with

```
# override dbt default of 80
max_line_length = 200
```

will avoid the error.

**Comment by peterbud** [comment]
With the latest `sqlfluff` there is no exception thrown with or without `max_line_length = 200`
